.github/workflows/ci.yml

```diff
@@ -241,27 +241,17 @@ jobs:
       if: matrix.crate == 'zerocopy' && matrix.features == '--all-features' && matrix.toolchain == 'nightly'
 
   kani:
-    runs-on: ubuntu-latest
-    name: Run tests under Kani
-    needs: generate_cache
+    on:
+      pull_request:
+      push:
     steps:
-      - uses: actions/checkout@8ade135a41bc03ea155e62e844d188df1ea18608 # v4.1.0
-      - name: Configure ZC_TOOLCHAIN environment variable
-        run: |
-          set -eo pipefail
-          echo "ZC_TOOLCHAIN=$(./cargo.sh --version nightly)" >> $GITHUB_ENV
-      - name: Install nigtly Rust toolchain (${{ env.ZC_TOOLCHAIN }})
-        uses: dtolnay/rust-toolchain@00b49be78f40fba4e87296b2ead62868750bdd83 # stable
+      name: 'Run Kani on your code.'
+      run-kani:
+        runs-on: ubuntu-latest        
+        uses: actions/checkout@8ade135a41bc03ea155e62e844d188df1ea18608 # v4.1.0    
+        uses: model-checking/kani-github-action@v1.0
         with:
-          toolchain: ${{ env.ZC_TOOLCHAIN }}
-      - run: |
-          set -eo pipefail
-          cargo install --locked kani-verifier
-          cargo kani setup
-          RUSTFLAGS="$ZC_NIGHTLY_RUSTFLAGS" ./cargo.sh +nightly kani                   \
-            --package zerocopy --all-features --output-format terse --randomize-layout \
-            --memory-safety-checks --overflow-checks --undefined-function-checks       \
-            --unwinding-checks
+          args: "--package zerocopy --all-features --output-format=terse --randomize-layout --memory-safety-checks --overflow-checks --undefined-function-checks --unwinding-checks"
```

Comment:
A few more changes, and then this should be good!
- Can you move the `name` field to the top level?
- Can you use the current name for this job ("Run tests under Kani")?
- Can you remove the `on:` directive?
- Can you move the `runs-on:` directive to the top level?

In total, with those changes, it should look like this:

```yaml
  kani:
    runs-on: ubuntu-latest
    name: Run tests under Kani
    steps:
      - uses: actions/checkout@8ade135a41bc03ea155e62e844d188df1ea18608 # v4.1.0
      - uses: model-checking/kani-github-action@v1.0
        with:
          args: "--package zerocopy --all-features --output-format=terse --randomize-layout --memory-safety-checks --overflow-checks --undefined-function-checks --unwinding-checks"
```